app/background/wallet/service.js

```diff
@@ -185,6 +185,40 @@ class WalletService {
     return res;
   };
 
+  updateAccountDepth = async (changeDepth, receiveDepth) => {
+    if (!this.name) return null;
+
+    await this._ensureClient();
+    const wallet = await this.node.wdb.get(this.name);
+
+    if (!wallet) return null;
+
+    const account = await wallet.getAccount('default');
+    const initChange = 0;
+    const initReceive = 0;
+    const b = this.node.wdb.db.batch();
+
+    account.changeDepth = changeDepth;
+    account.receiveDepth = receiveDepth;
+    await account.save(b);
+
+    if (changeDepth) {
+      for (let i = initChange; i < changeDepth; i++) {
+        const key = account.deriveChange(i);
+        await account.saveKey(b, key);
```

Comment:
So I think this is missing something actually: if you look at `account.syncDepth()` it also sets the depth in the account metadata which then gets written to the DB. If I'm reading this correctly, what this means is that although the wallet may have indeed saved all the addresses to the database, it still *thinks* its depth is:

`latest address index + lookahead` 

I'm not sure what the effect of this will be, it might just mean that the wallet will try to derive and save new addresses (that it already knows about, so no problem) up until the user-selected account depth actually matches the wallets internal account depth. In other words, the walletDB may have all the address, but the wallet/account object doesn't "know" that.

When users ask Bob for a new receive address, they will probably get the address at the wallet's internal depth index, which may be an address they have already used.

Reply:
Ok I see that actually in https://github.com/kyokan/bob-wallet/pull/301/commits/646164ff3758fc326f5b006e8ed03639f7da4419 is where the accounts depths are being set. So, this is bad because then users are going to put in a high number like 10,000 and end up giving the 10,000th address away for payment, and then not realize they will always need to generate 10,000 addresses every time they recover the wallet.